stonecrop/src/plugins/index.ts

```diff
@@ -1,9 +1,137 @@
 import { App, type Plugin } from 'vue'
+import type { RouteLocationNormalized } from 'vue-router'
+import { List, Map } from 'immutable'
 
+import DoctypeMeta from '../doctype'
 import Registry from '../registry'
 import { Stonecrop } from '../stonecrop'
 import type { InstallOptions } from '../types'
 
+/**
+ * Setup doctype metadata and load all records for the doctype
+ */
+async function setupDoctypeData(
+	registry: Registry,
+	stonecrop: Stonecrop,
+	doctype: string,
+	actualDoctype?: string,
+	apiBaseUrl = '/api'
+): Promise<void> {
+	try {
+		const targetDoctype = actualDoctype || doctype
+		// For list views, we need the list metadata which uses the '-list' suffix
+		const listDoctypeKey = targetDoctype.endsWith('-list') ? targetDoctype : `${targetDoctype}-list`
```

Comment:
I thought we were going to use a separate doctype for this instead of appending "list" to the doctype name